test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,376 @@
+                key = random_generator::instance().generate_random_string(tc->key_size);
+
+                /* Insert a key value pair. */
+                if (tc->insert(cc.cursor, cc.coll.id, key)) {
+                    if (tc->transaction.can_commit())
+                        /* We are not checking the result of commit as it is not necessary. */
+                        tc->transaction.commit();
+                } else {
+                    tc->transaction.rollback();
+                }
+
+                /* Sleep the duration defined by the configuration. */
+                tc->sleep();
+            }
+
+            /* Rollback any transaction that could not commit before the end of the test. */
+            if (tc->transaction.active())
+                tc->transaction.rollback();
+
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cc.cursor->reset(cc.cursor.get()));
+            if (++counter == collections_per_thread)
+                counter = 0;
+            testutil_assert(counter < collections_per_thread);
+        }
+    }
+
+    void
+    read_operation(test_harness::thread_context *tc) override final
+    {
+        /*
+         * Each read operation performs search_near calls with and without prefix enabled on random
+         * collections. Each prefix is randomly generated. The result of the seach_near call with
+         * prefix enabled is then validated using the search_near call without prefix enabled.
+         */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        std::map<uint64_t, scoped_cursor> cursors;
+        // TODO - Should this be a fixed value ? Should it be from the configuration ?
+        const uint64_t prefix_size = 3;
+        const char *key_prefix, *key_default;
+        int exact_prefix, exact_default;
+        int ret;
+
+        while (tc->running()) {
+            /* Get a random collection to work on. */
+            collection &coll = tc->db.get_random_collection();
+
+            /* Find a cached cursor or create one if none exists. */
+            if (cursors.find(coll.id) == cursors.end()) {
+                cursors.emplace(
+                  coll.id, std::move(tc->session.open_scoped_cursor(coll.name.c_str())));
+                auto &cursor_prefix = cursors[coll.id];
+                /* The cached cursors have the prefix configuration enabled. */
+                testutil_check(
+                  cursor_prefix.get()->reconfigure(cursor_prefix.get(), "prefix_key=true"));
+            }
+
+            auto &cursor_prefix = cursors[coll.id];
+
+            /*
+             * Select a random timestamp between the oldest and now and start the transaction at
+             * that time.
+             */
+            wt_timestamp_t ts = random_generator::instance().generate_integer(
+              tc->tsm->get_oldest_ts(), tc->tsm->get_next_ts());
```

Comment:
I was personally having troubles with the values that this was randomly generating. In my testing what was happening was only the upper 32 bits were set for each timestamp. This means we can potentially generate random integers for the lower 32 bits, creating more possibilities for the random generator to have an integer in bounds of the lower 32 bits. This wouldn't be useful for testing visibility of keys.  For example assume size is 32 bit (and we follow the same scenario) ts1 = `0x0001 0000`, ts2 = `0x0002 0000`, ts3 = `0x0003 0000`.  Overall if there were infinite attempts, it would lead to 50%. However if there are limited attempts like ours, it is more possible for the generator to only target numbers between ts1 and ts2.